Products/CMFPlone/browser/recyclebin.py

```diff
@@ -0,0 +1,592 @@
+        return {
+            "date_desc": "Newest First (default)",
+            "date_asc": "Oldest First",
+            "title_asc": "Title (A-Z)",
+            "title_desc": "Title (Z-A)",
+            "type_asc": "Type (A-Z)",
+            "type_desc": "Type (Z-A)",
+            "path_asc": "Path (A-Z)",
+            "path_desc": "Path (Z-A)",
+            "size_asc": "Size (Smallest First)",
+            "size_desc": "Size (Largest First)",
+        }
+
+    def get_clear_url(self, param_to_remove):
+        """Generate a URL that clears a specific filter parameter while preserving others
+
+        Args:
+            param_to_remove: The parameter name to remove from the URL
+
+        Returns:
+            URL string with the specified parameter removed
+        """
+        base_url = f"{self.context.absolute_url()}/@@recyclebin"
+        params = []
+
+        # Add search query if it exists and is not being removed
+        if param_to_remove != "search_query" and self.get_search_query():
+            params.append(f"search_query={self.get_search_query()}")
+
+        # Add filter type if it exists and is not being removed
+        if param_to_remove != "filter_type" and self.get_filter_type():
+            params.append(f"filter_type={self.get_filter_type()}")
+
+        # Add sort option if it exists, is not default, and is not being removed
+        sort_option = self.get_sort_option()
+        if param_to_remove != "sort_by" and sort_option != "date_desc":
+            params.append(f"sort_by={sort_option}")
+
+        # Construct final URL
+        if params:
+            return f"{base_url}?{'&'.join(params)}"
+        return base_url
+
+    def get_available_types(self, items):
+        """Get a list of all content types present in the recycle bin"""
+        types = set()
+        for item in items:
+            item_type = item.get("type")
+            if item_type:
+                types.add(item_type)
+        return sorted(list(types))
+
+    def get_items(self):
+        """Get all items in the recycle bin"""
+        recycle_bin = self.get_recycle_bin()
+        items = recycle_bin.get_items()
+
+        # Create a list of all items that are children of a parent in the recycle bin
+        child_items_to_exclude = []
+        for item in items:
+            # If this item is a parent with children, add its children to exclusion list
+            if "children" in item:
+                for child_id in item.get("children", {}):
+                    child_items_to_exclude.append(child_id)
+
+        logger.debug(f"Child items to exclude: {child_items_to_exclude}")
+
+        # Only include items that are not children of other recycled items
+        items = [item for item in items if item.get("id") not in child_items_to_exclude]
+
+        # For comments, add extra information about the content they belong to
+        for item in items:
+            if item.get("type") == "Discussion Item":
```

Comment:
```suggestion
            if item.get("type") == "Discussion item":
```

Reply:
changing this might cause issues, as here we are comparing with a predefined `Discussion Item` from `p.a.discussion`

Reply:
Ah, yes, sorry, I was overzealous.

Reply:
nevermind! 😅